src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -672,15 +680,56 @@ def image_mask_iterator():
                     description="1 if ROI was rejected or 0 if accepted as a cell during segmentation operation",
                 ),
             ],
-            id=roi_ids,
         )
+        if mask_type != "pixel" and mask_type != "voxel":  # roi_ids will not be written row-wise
+            plane_segmentation_kwargs.update(id=roi_ids)
+
         plane_segmentation = PlaneSegmentation(**plane_segmentation_kwargs)
+
         if include_roi_centroids:
             tranpose_roi_location_axes = (1, 0) if len(segmentation_extractor.get_image_size()) == 2 else (1, 0, 2)
             roi_locations = segmentation_extractor.get_roi_locations()[tranpose_roi_location_axes, :].T
             plane_segmentation.add_column(
                 name="ROICentroids", description="The x, y, (z) centroids of each ROI.", data=roi_locations
             )
+
+        if mask_type is not "None" and mask_type == "image":
+            plane_segmentation.add_column(
+                name="image_mask",
+                description="Image masks for each ROI.",
+                data=H5DataIO(DataChunkIterator(image_mask_iterator(), **iterator_options), **compression_options),
```

Comment:
I noticed here we may wish to do away with the iterator - the data as returned by ROIExtractors is an in-memory array, so there's no need for any iteration (though a `SliceableIterator` would still allow better default chunk shaping).

Reply:
This'll be a lesser priority to fix - outside of this PR

Reply:
Actually, I guess I did have to fix it here since the iterator wasn't operating with this approach (the original approach of directly column attachment bypasses a lot of the PyNWB API, for better or worse).

Wound up just removing it entirely because the underlying `_image_mask` attribute is already in memory for current formats so it's not even doing anything here. That said, this is an area of ROIExtractors that could really be improved